src/pg_to_evalscript/javascript_processes/mask.js

```diff
@@ -0,0 +1,60 @@
+function mask(arguments) {
+  const { data, mask, replacement = null } = arguments;
+
+  validateParameter({
+    processName: "mask",
+    parameterName: "data",
+    value: data,
+    required: true,
+  });
+
+  validateParameter({
+    processName: "mask",
+    parameterName: "mask",
+    value: mask,
+    required: true,
+  });
+
+  validateParameter({
+    processName: "mask",
+    parameterName: "replacement",
+    value: replacement,
+    required: false,
+    allowedTypes: ["number", "boolean", "string"],
+  });
+
+  for (let maskDim of mask.dimensions) {
+    const dataDim = data.getDimensionByName(maskDim.name);
+
+    if (!dataDim) {
+      throw new Error(`Dimension \`${maskDim.name}\` from argument \`mask\` not in argument \`data\`.`);
+    }
+
+    if (maskDim.type !== dataDim.type) {
+      throw new Error(`Type of the dimension \`${maskDim.name}\` from argument \`mask\` is not the same as in argument \`data\`.`);
+    }
+
+    if (maskDim.labels.length !== dataDim.labels.length || !maskDim.labels.every((l) => dataDim.labels.includes(l))) {
+      throw new Error(`Labels for dimension \`${maskDim.name}\` from argument \`mask\` are not the same as in argument \`data\`.`);
+    }
+  }
+
+  let newData = data.clone();
+  let newDataFlat = newData.flattenToArray();
+
+  const maskBroadcasted = broadcastNdarray(mask.data, newData.data.shape);
+  const maskBroadcastedFlat = flattenToNativeArray(maskBroadcasted);
+
+  const replacement_val = replacement === undefined ? null : replacement;
+
+  for (let i = 0; i < newDataFlat.length; i++) {
```

Comment:
Of course the approach with flattening the arrays and comparing values will only work if the dimensions will be in the same order in the `DataCube`. If the dimensions would be switched, the values that are being masked here have different labels than the mask value. 

I would say that in the majority of the cases the dimension order will be the same, but this should either work with dimensions in arbitrary order or raise and error of order doesn't match.

Reply:
I realized that this problem and the problem mentioned above are connected, so I replied in the comment above.